### PR TITLE
core/types, params: add blob transaction type, RLP encoded for now

### PR DESCRIPTION
**Pull request by karalabe**
This PR is another subset of 4844. The purpose is to introduce the new transaction type for blobs, extend the generic transaction interface with the getters needed by blob txs and to add the signer that can sign and recover these new transaction types.

For now the blob txs use RLP, since SSZ integration will be a different step where the scope should be formatting, not type extensions.

### CODE REVIEW

File: core/types/transaction.go [hunk 1]
```diff
@@ -42,9 +42,10 @@ var (
 
 // Transaction types.
 const (
-	LegacyTxType = iota
-	AccessListTxType
-	DynamicFeeTxType
+	LegacyTxType     = 0x00
+	AccessListTxType = 0x01
+	DynamicFeeTxType = 0x02
+	BlobTxType       = 0x05
```
Comment (karalabe) [inline core/types/transaction.go:48]: This is super weird, can we please have 0x04 and have L2s fix their shit?

Comment (rjl493456442) [reply core/types/transaction.go:48]: What's 0x03 though?

Comment (karalabe) [reply core/types/transaction.go:48]: I think they used non-eip-155 txs, so they didn't include the chain id. Or something werid like that. IMHO that's up on them then.

Comment (rjl493456442) [reply core/types/transaction.go:48]: Hmmm, it's really bad design IMHO. But you are right, it's up to them anyway. This transaction type definition shared by all different networks is a bit ugly I really want to say.

**Review by Mholt74 — APPROVED** [review]

**Comment by Skylerspencer** [comment]
I still  have copies of your scams and cover up its best you stop and just
let this court case between me and those guys Number one BNB the ring
leader . You guys don't have enough  info your being used to commit crime
not them can't talk to you guys anymore counsel good by.

**Comment by Mholt74** [comment]
****

**Comment by Szybka22** [comment]
gh pr checkout 27087

**Review by holiman — APPROVED** [review]
LGTM. Seems most changes are not related to blob-txs per se, but rather making the signer timestamp-aware.

File: core/types/tx_blob.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	V *uint256.Int `json:"v" gencodec:"required"`
+	R *uint256.Int `json:"r" gencodec:"required"`
+	S *uint256.Int `json:"s" gencodec:"required"`
+}
+
+// copy creates a deep copy of the transaction data and initializes all fields.
+func (tx *BlobTx) copy() TxData {
+	cpy := &BlobTx{
+		Nonce: tx.Nonce,
+		To:    tx.To, //copyAddressPtr(tx.To),
```
[the remark this answers is not in the document]
Comment (karalabe) [reply core/types/tx_blob.go:51]: It's a refactor bug. The exact fields are a bit fluid and it wasn't clear due to SSZ union debates whether we should allow contract creation or not. My working code uses non-pointers and I reverted back to pointer for this PR to keep it flexible, but the type change didn't blow on compile. Should be copyAddressPtr

[the remark this answers is not in the document]
Comment (karalabe) [reply core/types/tx_blob.go]: Fixed, PTAL

File: core/state_processor.go [hunk 3]
```diff
@@ -74,7 +74,7 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	vmenv := vm.NewEVM(blockContext, vm.TxContext{}, statedb, p.config, cfg)
 	// Iterate over and process the individual transactions
 	for i, tx := range block.Transactions() {
-		msg, err := TransactionToMessage(tx, types.MakeSigner(p.config, header.Number), header.BaseFee)
+		msg, err := TransactionToMessage(tx, types.MakeSigner(p.config, header.Number, header.Time), header.BaseFee)
```
[the remark this answers is not in the document]
Comment (karalabe) [reply core/state_processor.go]: Good suggestion, fixed, PTAL

File: core/types/transaction.go (hunk 1 above)
Comment (karalabe) [reply core/types/transaction.go]: Was fixed to be 0x03.

**Pull request merged**